### PR TITLE
Fix SchemaRegistry live tests

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes SchemaRegistry live tests from failing as they we attempting to set a recording policy during `AZURE_TEST_MODE=LIVE` which resulted in an exception being thrown.

### CODE REVIEW

File: sdk/schemaregistry/azure-data-schemaregistry-apacheavro/src/test/java/com/azure/data/schemaregistry/apacheavro/SchemaRegistryApacheAvroSerializerIntegrationTest.java [hunk 1]
```diff
@@ -62,29 +53,19 @@ public class SchemaRegistryApacheAvroSerializerIntegrationTest extends TestProxy
...
     private String eventHubName;
     private String connectionString;
 
     @Override
     protected void beforeTest() {
+        String endpoint;
+        TokenCredential tokenCredential;
         if (interceptorManager.isPlaybackMode()) {
-            tokenCredential = mock(TokenCredential.class);
+            tokenCredential = new MockTokenCredential();
```
Comment (conniey) [inline sdk/schemaregistry/azure-data-schemaregistry-apacheavro/src/test/java/com/azure/data/schemaregistry/apacheavro/SchemaRegistryApacheAvroSerializerIntegrationTest.java:66]: What was the cause of the test failures? Was mockito incorrectly mocking this?

Comment (alzimmermsft) [reply sdk/schemaregistry/azure-data-schemaregistry-apacheavro/src/test/java/com/azure/data/schemaregistry/apacheavro/SchemaRegistryApacheAvroSerializerIntegrationTest.java:66]: The mock was fine, this is just a better version than the mock as we don't care about the credential response during playback other than it didn't throw an error. This change was just a piece of reducing places where we use Mockito as there is a lot of baggage with it, especially given Mockito is no longer actively developing with a Java 8 baseline.

**Review by conniey — APPROVED** [review]
LGTM. Can you update the description explaining what the fix was/why it was failing?

**Pull request merged**